### ISSUE TITLE
Feat: allow Select to accept value, label pairs

### ISSUE THREAD

**Issue by mangecoeur** [widgetti/solara#186]
Ipywidgets Select allows to provide (value, label) pairs instead of just label, which is quite useful for selecting keyed data. Vue instead allows to provide a list[dict] and item_name, item_value keys to use from that dict. it would be nice to support one of these (maybe ipywidgets one is most practical)

**Comment by maartenbreddels** [comment]
Yes I agree. Like ipywidgets, we should also allow the value to be a Python value (so the only information that goes to the frontend is an index).
